tests/png/bug00295.c

```diff
@@ -0,0 +1,39 @@
+/**
+ * Regression test for <https://github.com/libgd/libgd/issues/295>.
+ *
+ * We're testing that reading a PNG which might trigger a libpng warning
+ * does not trigger a GD_ERROR.
+ */
+
+
+#include "gd.h"
+#include "gd_errors.h"
+#include "gdtest.h"
+#include <string.h>
+
+
+static void error_handler(int priority, const char *format, va_list args)
+{
+    char msg[256];
+
+    vsprintf(msg, format, args);
+    gdTestAssertMsg(priority != GD_ERROR,
+                    "unexpected GD_ERROR: %s\n", msg);
+}
+
+
+int main()
+{
+    gdImagePtr im;
+    FILE *fp;
+
+    gdSetErrorMethod(error_handler);
+    fp = gdTestFileOpen2("png", "bug00295.png");
+
+    im = gdImageCreateFromPng(fp);
+
+    fclose(fp);
+    gdImageDestroy(im);
+
+    return gdNumFailures();

```

Comment:
what guarantees that `error_handler` was called ?  should we have a global boolean like `called` that starts off `false` and `error_handler` sets to `true` ?